packages/amplify-graphql-model-transformer/src/resolvers/rds/mutation.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  comment,
+  compoundExpression,
+  Expression, list,
+  methodCall,
+  obj,
+  printBlock,
+  qref,
+  ref,
+  set,
+  str,
+  toJson,
+} from 'graphql-mapping-template';
+import { ModelDirectiveConfiguration } from '../../directive';
+
+/**
+ * Generate mapping template that sets default values for create mutation
+ * @param modelConfig directive configuration
+ */
+export const generateCreateInitSlotTemplate = (modelConfig: ModelDirectiveConfiguration): string => {
+  const statements: Expression[] = [
+    // initialize defaultValues
+    qref(
+      methodCall(
+        ref('ctx.stash.put'),
+        str('defaultValues'),
+        methodCall(ref('util.defaultIfNull'), ref('ctx.stash.defaultValues'), obj({})),
+      ),
+    ),
+  ];
+
+  if (modelConfig?.timestamps) {
+    statements.push(set(ref('createdAt'), methodCall(ref('util.time.nowISO8601'))));
+    statements.push(qref(methodCall(ref('ctx.stash.defaultValues.put'), str('id'), methodCall(ref('util.autoId')))));
+    if (modelConfig.timestamps.createdAt) {
+      statements.push(qref(methodCall(ref('ctx.stash.defaultValues.put'), str(modelConfig.timestamps.createdAt), ref('createdAt'))));
+    }
+    if (modelConfig.timestamps.updatedAt) {
+      statements.push(qref(methodCall(ref('ctx.stash.defaultValues.put'), str(modelConfig.timestamps.updatedAt), ref('createdAt'))));
+    }
+  }
+  statements.push(
+    toJson(
+      obj({
+        version: str('2018-05-29'),
+        payload: obj({}),
+      }),
+    ),
+  );
+  return printBlock('Initialization default values')(compoundExpression(statements));
+};
+
+export const generateLambdaCreateRequestTemplate = (tableName: string, operationName: string): string => {
+  return printBlock('Invoke RDS Lambda data source')(
+    compoundExpression([
+      set(ref('lambdaInput'), obj({})),
+      set(ref('lambdaInput.table'), str(tableName)),
+      set(ref('lambdaInput.args'), obj({})),
+      set(ref('lambdaInput.operation'), str('CREATE')),
+      set(ref('lambdaInput.operationName'), str(operationName)),
+      comment('Set the default values to put request'),
+      set(ref('lambdaInput.args.input'), methodCall(ref('util.defaultIfNull'), ref('ctx.stash.defaultValues'), obj({}))),
+      comment('copy the values from input'),
+      qref(methodCall(ref('lambdaInput.args.input.putAll'), methodCall(ref('util.defaultIfNull'), ref('context.arguments.input'), obj({})))),
```

Comment:
Did we check how values get passed in case of using variables?

Reply:
Possibly we can pass the full context to the lambda and extract the values from there.

Reply:
I may leave this as a followup optimization just because of the size of scope